k8s/migration/pkg/utils/credutils.go

```diff
@@ -561,10 +561,18 @@ func GetAllVMs(ctx context.Context, scope *scope.VMwareCredsScope, datacenter st
 	}
 	// Pre-allocate vminfo slice with capacity of vms to avoid append allocations
 	vminfo := make([]vjailbreakv1alpha1.VMInfo, 0, len(vms))
+
+	// Create a semaphore to limit concurrent goroutines
+	semaphore := make(chan struct{}, constants.VCenterVMScanConcurrencyLimit)
+
 	for i := range vms {
+		// Acquire semaphore (blocks if 100 goroutines are already running)
+		semaphore <- struct{}{}
 		wg.Add(1)
 		go func(i int) {
 			defer wg.Done()
+			// Release semaphore when done
```

Comment:
@OmkarDeshpande7 One more nit: can u pass vm variable in this go routine to avoid any race condition ?

Reply:
There is an inherent issue with golang for loop and pointers as reference variables. In this case I have passed the iterator and accessing the right value with vms[i]. To avoid the race condition.

Reply:
@OmkarDeshpande7 I  fear this

for i := range vms {
	i := i // capture loop variable
	go func() {
		_ = vms[i] // SAFE
	}()
}
Without i := i, you'll likely get a race where all goroutines refer to the same final value of i.

Reply:
We are sending a local copy of i to each goroutine as a function paramater. So loop variable is captured